proxy/http2/Http2ClientSession.cc

```diff
@@ -543,6 +543,11 @@ Http2ClientSession::state_process_frame_read(int event, VIO *vio, bool inside_fr
   }
 
   while (this->sm_reader->read_avail() >= (int64_t)HTTP2_FRAME_HEADER_LEN) {
+    // Cancel reading if there was an error
+    if (connection_state.tx_error_code.code != static_cast<uint32_t>(Http2ErrorCode::HTTP2_ERROR_NO_ERROR)) {
```

Comment:
When graceful shutdown is started, `connection_state.tx_error_code` will be `{ProxyErrorClass::SSN, Http2ErrorCode::HTTP2_ERROR_NO_ERROR}`. Should we also cancel reading?

Reply:
No, I don't think so. Because we still need to read WINDOW_UPDATE in that case.

Reply:
Fair enough.